File: add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
  # convert the linked lists to integers
  def conv_int(l: ListNode):
    s = ""
    while l != None:
      s += str(l.val)
      l = l.next
    return int(s[::-1])
    
  # convert sum of integers to a linked list
  def to_list(n: int):
    s = str(n)[::-1]
    head = prev = None
    for ch in s:
      node = ListNode(int(ch))
      if prev is not None:
        prev.next = node
        prev = node
      if head is None:
        head = prev
    return head
    
  # return the sum of the two linked lists
  def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
    a = Solution.conv_int(l1)
    b = Solution.conv_int(l2)
    return Solution.to_list(a + b)
        

File: add-two-numbers/add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
  let sum = 0;
  let curr = new ListNode(0);
  let res = curr;

  while (l1 || l2) {
    if (l1) {
      sum += l1.val;
      l1 = l1.next;
    }
    if (l2) {
      sum += l2.val;
      l2 = l2.next;
    }

    curr.next = new ListNode(sum % 10);
    curr = curr.next;

    sum = sum > 9 ? 1 : 0;
  }

  if (sum) {
    curr.next = new ListNode(sum);
  }

  return res.next;
};
